src/Microsoft.Diagnostics.Monitoring.Options/MonitorCapability.cs

```diff
@@ -1,18 +1,22 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System.ComponentModel.DataAnnotations;
 using System.Text.Json.Serialization;
 
 namespace Microsoft.Diagnostics.Monitoring.Options
 {
-    public class MonitorCapability
+    public class MonitorCapability : IMonitorCapability
     {
-        [Required]
         [JsonPropertyName("name")]
-        public string Name { get; set; } = string.Empty;
+        public string Name { get; init; }
```

Comment:
Can the init be removed from both of the properties since they are set through the constructor? I think STJ should still include them according to https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializeroptions.ignorereadonlyproperties?view=net-9.0